tests/local/network.go

```diff
@@ -226,24 +265,25 @@ func (n *localNetwork) setSubnetValues(subnetID ids.ID) {
 	// TeleporterRegistryAddress is set in DeployTeleporterRegistryContracts
 }
 
-// deployTeleporterContracts deploys the Teleporter contract to all subnets.
+// DeployTeleporterContracts deploys the Teleporter contract to all subnets.
 // The caller is responsible for generating the deployment transaction information
-func (n *localNetwork) deployTeleporterContracts(
+func (n *LocalNetwork) DeployTeleporterContracts(
 	transactionBytes []byte,
 	deployerAddress common.Address,
 	contractAddress common.Address,
 	fundedKey *ecdsa.PrivateKey,
 ) {
 	log.Info("Deploying Teleporter contract to subnets")
 
-	subnetsInfoList := n.GetSubnetsInfo()
-
 	// Set the package level teleporterContractAddress
 	n.teleporterContractAddress = contractAddress
 
 	ctx := context.Background()
 
-	for _, subnetInfo := range subnetsInfoList {
+	subnets := n.GetSubnetsInfo()
+	subnets = append(subnets, *n.primaryNetworkInfo)
```

Comment:
Maybe worth considering something along the lines of a `network.GetAllBlockchainInfo` helper for cases like this.

Reply:
Good call, added